src/olympia/devhub/views.py

```diff
@@ -1490,32 +1490,33 @@ def submit_finish(request, addon_id, addon):
     # Bounce to the versions page if they don't have any versions.
     if not addon.versions.exists():
         return redirect(addon.get_dev_url('versions'))
-    sp = addon.current_version.supported_platforms
-    is_platform_specific = sp != [amo.PLATFORM_ALL]
+    uploaded_version = addon.versions.latest()
```

Comment:
If a developer purposely went back to that step after submission (e.g. went to developers/addon/addon-slug/submit/) then `.latest` would include the most recent version, regardless of status, whereas `.find_latest_version` wouldn't include a disabled version.  Neither really makes much sense tbh so it's probably not worth shaving that yak.